lib/jekyll-paginate-v2/generator/paginationPage.rb

```diff
@@ -9,10 +9,14 @@ module PaginateV2::Generator
     # This page exists purely in memory and is not read from disk
     #
     class PaginationPage < Page
+      attr_reader :relative_path
+
       def initialize(page_to_copy, cur_page_nr, total_pages, index_pageandext)
         @site = page_to_copy.site
         @base = ''
-        @url = ''
+        @url  = ''
+        @relative_path = page_to_copy.relative_path
```

Comment:
I don´t understand, adding an attribute which is not used anywhere feels unnecessary...

Reply:
> which is not used anywhere

Actually, `:relative_path` is used multiple times in a build cycle:
  - to compute a `Page` instance's [`path`](https://github.com/jekyll/jekyll/blob/b586bed0ee447a2f9e09fed7f752118c5a495a95/lib/jekyll/page.rb#L141-L143) which is in turn passed to a Liquid template that is accessed via `{{ page.path }}`
   - in [`Regenerator`](https://github.com/jekyll/jekyll/blob/b586bed0ee447a2f9e09fed7f752118c5a495a95/lib/jekyll/regenerator.rb#L168)
   - multiple times in [`Renderer`](https://github.com/jekyll/jekyll/blob/b586bed0ee447a2f9e09fed7f752118c5a495a95/lib/jekyll/renderer.rb#L78)

Reply:
excerpt from building `examples/01-typicalblog`  with `--verbose` option
```
         Rendering: feed.xml
  Pre-Render Hooks: feed.xml
  Rendering Liquid: feed.xml
  Rendering Markup: feed.xml
  Rendering Layout: feed.xml
         Rendering: index.html
  Pre-Render Hooks: index.html
  Rendering Markup: index.html
  Rendering Layout: index.html
     Layout source: site
         Rendering: .html
  Pre-Render Hooks: .html
  Rendering Markup: .html
  Rendering Layout: .html
     Layout source: site
```
The `.html` towards the end is from `page 2` pagination page that got rendered into `_site/page/2/index.html`..

Reply:
Ah OK, I understand